fix: Change non-existent secrets dir into a warning (#2265)
samuelcolvin/pydantic: changes/2175-davidolrik.md

> fix: Change non-existent secrets dir from exception to warning

Reviewer: ```suggestion
Allow non-existent secrets directories by issuing only a warning
```

Author: ```suggestion
Allow non-existent secrets directory by only issuing a warning
```
